### PR TITLE
seed warning for distributed workflows

### PR DESCRIPTION
**Pull request by nec4**
# PR Checklist

- [] Bug fix
- [x] Feature addition/change
- [x] Documentation addition/change
- [] Test addition/change
- [] Black formatting

---

Hello. This PR adds a warning when users are setting `seed_everything` to `false`/`null` in configs for distributed workflows. Seems that this sort of a hidden corner issue/pain with PyTorch Lightning, as the official `LightningDataModule` docs/examples fail to underscore the following issue:

```
The `setup()` method is called per rank/procid. This means that for DDP, the whole dataset is sent to each GPU/device (on each node if there is multinode training) and they can run their own (self-seeded) random number generators when making data splits if `seed_everything` is set to `false`/`null`. Effectively, this will lead to unintentional data leakage between train/val/test sets if the user does not read any predetermined splits from disk.
```

I have tested this on a single node (all I have access to at the moment) with 4 A100s. I can confirm that for the `ChignolinDataset` example, if `seed_everything` is set properly (ie, any positive integer), each rank/procid splits the data into the same train/test/val partitions properly. However, if `seed_everything` is set to `false`/`null`, this is no longer true. Also it should be noted that, while if completely unspecified, `seed_everything` defaults to the integer value `0`, nothing prevents a user with alternative intuition from setting it to `null`/`false` in the config and leading themselves to this issue.

This issue was brought up in the recent-ish past with the Lightning developers, and they acknowledged the potential issue but basically say "always set `seed_everything` for distributed workflows, or predefine the splits and load them from disk" (also an option we provide with `mlcg`). Just as well, `prepare_data` is not so helpful here because its not intended to be used for setting state. See here to unwrap my paraphrasing: https://github.com/Lightning-AI/pytorch-lightning/issues/18638

In the end, I have opted to follow the Lightning devs thinking and just add a warning to alert users who may hit this corner case, as setting `seed_everything` seems to solve this potential issue.

A couple questions:

1. Is a simple warning enough (eg, warning fatigue with all the other stock torch/pl warnings)?
2. Right now the warning is only for the cli training. Is it necessary to go beyond this? I'm not sure if anyone is using non-cli Lightning pipelines with `mlcg` atm.

Still have to check a few things but happy for feedback.

✌️

### CODE REVIEW

**Comment by sayeg84** [comment]
Thanks for the PR. I will give it a test early next week and, if everything works, merge it. To answer your questions:

1. I think the warning is enough, even if there are too many already, as throwing an error is too excessive. 
2. The only non-cli workflows are the example notebook and related things, for which people shouldn't be using more than 1 device. I think its safe to put the fix only for the CLI cases.

**Comment by nec4** [comment]
Thanks! I also just now tested it across multiple nodes and the situation is the same: setting `seed_everything` to an `int` gives the same splits across the ranks, while setting it to `false/null` does not.

**Comment by sayeg84** [comment]
There were some bugs due to not using the right config object and not parsing the parameters in the right type, which happens if you set the parameters from the command line instead of setting a yaml.

I also formatted the warning so that it displays better in a terminal and its more easy to spot.

Let me know if this looks fine to you and I will merge

**Comment by nec4** [comment]
Thanks for the additions and corrections - LGTM.

**Review by sayeg84 — APPROVED** [review]
